### ISSUE TITLE
Required fields in SchemaObject are rendered incorrectly

### ISSUE THREAD

**Issue by fuzza** [asyncapi/docgen#27]
It seems that there is an issue with rendering of required properties of SchemaObject. E.g. for a following API schema `first_field` of `testSchema` won't be rendered as a required.

```yaml
asyncapi: "1.1.0"
info:
  title: Required fields issue demo
  version: "0.0.1"

topics:
  test:
    publish:
      $ref: "#/components/messages/testMessage"
      
components:
  messages:
    testMessage:
      summary: Testing
      payload:
        $ref: "#/components/schemas/testSchema"

  schemas:
    testSchema:
      type: object
      required:
        - first_field
      properties:
        first_field:
          type: string
        second_field:
          type: string
```

<img width="918" alt="screen shot 2018-06-21 at 4 08 06 pm" src="https://user-images.githubusercontent.com/2526427/41721475-a5248e48-756e-11e8-91ec-693da8d2cb94.png">

From what I see in`schema-prop` template, there is a check of required array existance inside the `prop` object while it needs to check that parent schema object has `propName` in `required` array. 

I have WIP fix for this issue in a forked repo, so please let me know if you want me to submit a PR.  Looking forward for you feedback.

**Comment by fmvilas** [comment]
Hi @fuzza. Thanks for reporting. Absolutely, your PR is more than welcome!

**Comment by NicoHood** [comment]
Was this fix already pushed to the online editor? I saw similar problems. If yes, please update if no there are still problems.

**Comment by fmvilas** [comment]
<img width="1277" alt="screen shot 2018-07-21 at 19 36 42" src="https://user-images.githubusercontent.com/242119/43038647-7cda820a-8d1d-11e8-83c7-e71d7d660b26.png">

@NicoHood @fuzza it's already working on the Editor. I also published the fix to npm (version 1.10.2): https://www.npmjs.com/package/asyncapi-docgen.

Thanks for the PR @fuzza!
